pkg/auth/tokenparser.go

```diff
@@ -64,27 +42,32 @@ func NewTokenParser(keyManager *KeyManager) (*TokenParser, error) {
 
 // FromString parses a JWT, validates the signature and returns the claims struct.
 func (tp *TokenParser) FromString(jwtEncoded string) (*TokenClaims, error) {
-	token, err := jwt.ParseWithClaims(jwtEncoded, &TokenClaims{}, func(token *jwt.Token) (interface{}, error) {
-		// validate the alg is what we expect
-		if _, ok := token.Method.(*jwt.SigningMethodRSA); !ok {
-			return nil, fmt.Errorf("unexpected signing method: %v", token.Header["alg"])
-		}
+	token, err := jwt.ParseWithClaims(
+		jwtEncoded,
+		&TokenClaims{},
+		func(token *jwt.Token) (interface{}, error) {
+			// validate the alg is what we expect
+			if _, ok := token.Method.(*jwt.SigningMethodRSA); !ok {
+				return nil, fmt.Errorf("unexpected signing method: %v", token.Header["alg"])
+			}
 
-		kid := token.Header["kid"]
-		if kid == nil {
-			return nil, errors.New("no key id given in the token")
-		}
-		kidStr, ok := kid.(string)
-		if !ok {
-			return nil, errors.New("given key id has unknown type")
-		}
-		// get the public key for kid from keyManager
-		publicKey, err := tp.keyManager.Key(kidStr)
-		if err != nil {
-			return nil, err
-		}
-		return publicKey, nil
-	})
+			kid := token.Header["kid"]
+			if kid == nil {
+				return nil, errors.New("no key id given in the token")
+			}
+			kidStr, ok := kid.(string)
+			if !ok {
+				return nil, errors.New("given key id has unknown type")
+			}
+			// get the public key for kid from keyManager
+			publicKey, err := tp.keyManager.Key(kidStr)
+			if err != nil {
+				return nil, err
+			}
+			return publicKey, nil
+		},
+		jwt.WithLeeway(leeway),
```

Comment:
This is the leeway option with make our parser to take the leeway window into account when parsing the token instead of relaying the Valid() interface above.